src/dmd/target.d

```diff
@@ -893,12 +902,45 @@ extern (C++) struct Target
     out(result) { assert(result || global.params.isWindows); }
     do
     {
-        return global.params.isLinux
-            || global.params.isOSX
-            || global.params.isFreeBSD
-            || global.params.isOpenBSD
-            || global.params.isDragonFlyBSD
-            || global.params.isSolaris;
+        with (Platform)
+        final switch (platform)
+        {
+            case Linux, OSX, FreeBSD, OpenBSD, DragonFlyBSD, Solaris:
+                return true;
+
+            case Windows:
+                return false;
+        }
+    }
+
+    /*
+     * Returns: list of describing names for the current platform instance.
+     */
+    extern (D) string[] platformNames()
```

Comment:
It would be more useful if this was extern(C++), or if it returned a field that could be set by the glue/backend.

Actually, here a field would even be preferable to avoid allocating on each call.